### ISSUE TITLE
validate non-empty string values on some resources

### ISSUE THREAD

**Issue by ryane** [asteris-llc/converge#337]
After #321, it looks like we have to add back some validation in Preparers to ensure that some string fields are not empty. 
- docker.container: name, image
- group: name
- user: username

These fields are required but they also should not be empty. I have confirmed that the docker.container fields are definitely a problem and it _looks_ like user and group will be a problem as well. For example, converge does not complain about this:

``` hcl
docker.container "nginx" {
  name  = "nginx"
  image = ""
}
```

but trying to apply this will obviously fail so we should add further validations.

**Comment by ryane** [comment]
#345 fixes the docker related validations

**Comment by BrianHicks** [comment]
If you use a pointer to a string as your destination type you can detect this now. Would that solve this problem?

Concretely: the `Image string` would go to `Image *string`. If the pointer is nil, it's not set. If it's blank, it's been explicitly set to blank.

**Comment by ryane** [comment]
this was resolved by #494
